cirq/contrib/qasm_import/_parser_test.py

```diff
@@ -368,39 +392,102 @@ def test_unknown_function():
         parser.parse(qasm)
 
 
-def test_u_gate_zero_params_error():
+rotation_gates = [
+    ('rx', cirq.Rx),
+    ('ry', cirq.Ry),
+    ('rz', cirq.Rz),
+]
+
+
+@pytest.mark.parametrize('qasm_gate,cirq_gate', rotation_gates)
+def test_rotation_gates(qasm_gate: str, cirq_gate: cirq.SingleQubitGate):
     qasm = """OPENQASM 2.0;
+     include "qelib1.inc";
+     qreg q[2];
+     {0}(pi/2) q[0];
+     {0}(pi) q;
+    """.format(qasm_gate)
+
+    parser = QasmParser()
+
+    q0 = cirq.NamedQubit('q_0')
+    q1 = cirq.NamedQubit('q_1')
+
+    expected_circuit = Circuit()
+    expected_circuit.append(cirq_gate(np.pi / 2).on(q0))
+    expected_circuit.append(
+        cirq.Moment([cirq_gate(np.pi).on(q0),
+                     cirq_gate(np.pi).on(q1)]))
+
+    parsed_qasm = parser.parse(qasm)
+
+    assert parsed_qasm.supportedFormat
+    assert parsed_qasm.qelib1Include
+
+    ct.assert_same_circuits(parsed_qasm.circuit, expected_circuit)
+    assert parsed_qasm.qregs == {'q': 2}
+
+
+@pytest.mark.parametrize('qasm_gate', [g[0] for g in rotation_gates])
+def test_rotation_gates_wrong_number_of_args(qasm_gate: str):
+    qasm = """
+     OPENQASM 2.0;
+     include "qelib1.inc";             
      qreg q[2];     
-     U() q[1];"""
+     {}(pi) q[0], q[1];     
+""".format(qasm_gate)
 
     parser = QasmParser()
 
-    with pytest.raises(QasmException,
-                       match=r"U takes 3 parameter\(s\).*got.*0.*line 3"):
+    with pytest.raises(
+            QasmException,
+            match=r".*{}.* takes 1 arg\(s\).*got.*2.*line 5".format(qasm_gate)):
```

Comment:
Your matches are often very specific. No need to change that. Personally I make them vague (e.g. just `"takes 1 arg"`) so that it's easier to update the language.

Reply:
Yeah it's a trade-off, here I wanted the data to show up - i.e lineno, num args accepted and actual arg number with some context to make the test understandable what I'm matching for. I could make the context smaller though. I'll think about it.